packages/liquidation-sdk-viem/examples/whitelisted-erc4626-1inch.ts

```diff
@@ -240,41 +241,159 @@ export const check = async <
                   }
                   // Default case, use 1inch/paraswap for other collaterals
                   default: {
-                    const bestSwap = await fetchBestSwap({
-                      chainId,
-                      src: srcToken,
-                      dst: market.params.loanToken,
-                      amount: srcAmount,
-                      from: executorAddress,
-                      slippage,
-                      includeTokensInfo: false,
-                      includeProtocols: false,
-                      includeGas: false,
-                      allowPartialFill: false,
-                      disableEstimate: true,
-                      usePermit2: false,
-                    });
-
-                    if (!bestSwap)
-                      throw Error(
-                        "could not fetch swap from both 1inch and paraswap",
-                      );
-
-                    dstAmount = BigInt(bestSwap.dstAmount);
-
-                    if (
-                      dstAmount < repaidAssets.wadMulDown(BigInt.WAD + slippage)
-                    )
-                      return;
-
-                    encoder
-                      .erc20Approve(srcToken, bestSwap.tx.to, srcAmount)
-                      .pushCall(
-                        bestSwap.tx.to,
-                        BigInt(bestSwap.tx.value),
-                        bestSwap.tx.data,
-                      );
-
+                    let retry = true;
+                    const tries: { srcAmount: bigint; srcToken: Address }[] =
+                      [];
+                    while (retry) {
```

Comment:
```suggestion
                    while (true) {
```

Reply:
With `break` statements

Ideally also you could export the whole `while` loop to a function so it is more readable (and maybe you can take advantage of early returns in this case